test/test_resource.rs

```diff
@@ -0,0 +1,24 @@
+use nix::sys::resource::{getrlimit, setrlimit, Resource};
+
+/// Tests the RLIMIT_NOFILE functionality of getrlimit(), where the resource RLIMIT_NOFILE refers
+/// to the maximum file descriptor number that can be opened by the process (aka the maximum number
+/// of file descriptors that the process can open, since Linux 4.5).
+///
+/// We first fetch the existing file descriptor maximum values using getrlimit(), then edit the
+/// soft limit to make sure it has a new and distinct value to the hard limit. We then setrlimit()
+/// to put the new soft limit in effect, and then getrlimit() once more to ensure the limits have
+/// been updated.
+#[test]
+pub fn test_resource_limits_nofile() {
+    let (soft_limit, hard_limit) = getrlimit(Resource::RLIMIT_NOFILE).unwrap();
+
+    // make sure the soft limit and hard limit are not equal
```

Comment:
What ensures that the soft limit is not equal to the hard limit?

Reply:
This is actually obsolete comments from the very early version of the test, see https://github.com/LMJW/nix/commit/4f8b468db6ff2388afc0051032acbc1e662afcf2

For this particular test, I believe we don't need to have soft limit to equal to hard limit